testutils/file_utils.go

```diff
@@ -18,6 +20,52 @@ import (
 	"go.viam.com/rdk/utils"
 )
 
+const osDarwin = "darwin"
+
+// BuildViamServer will attempt to build the viam-server (full static if on linux). If successful, this function will
+// return the path to the executable.
+func BuildViamServer(tb testing.TB) string {
+	tb.Helper()
+
+	command := "full-static"
+	if runtime.GOOS == osDarwin {
+		command = "server"
+	}
+
+	builder := exec.Command("make", command)
+	builder.Dir = utils.ResolveFile(".")
+	_, err := builder.CombinedOutput()
+	if err != nil {
+		tb.Error(err)
+	}
+	if tb.Failed() {
+		tb.Fatal("failed to build viam-server executable")
+	}
+	serverPath := ""
+	// look for the built executable
+	searchFunc := func(path string, dir fs.DirEntry, err error) error {
+		if err != nil {
+			return fs.SkipDir
+		}
+		if dir.IsDir() {
+			return nil
+		}
+		if strings.Contains(dir.Name(), "viam-server") {
+			serverPath = path
+			return fs.SkipAll
+		}
+		return nil
+	}
+	err = filepath.WalkDir(utils.ResolveFile("bin"), searchFunc)
```

Comment:
if I"m running tests on my local machine there are a lot of unpleasant things in that bin directory fyi